meshnets/data_processing/dataset_statistics.py

```diff
@@ -0,0 +1,49 @@
+"""File to compute dataset statistics"""
+import collections.abc
+
+import numpy as np
+import npstreams
+
+
+def _yield_array(array, axis):
+    if axis is None:
+        for number in array.flatten():
+            yield number
+    else:
+        yield array
+
+
+def make_iterator(dataset, key, axis, max_iterations):
+    """Make iterator over dataset"""
+    if isinstance(dataset, collections.abc.Iterable):
+        for i, example in enumerate(dataset):
+            if i >= max_iterations:
+                break
+            array = np.array(example[key])
+            yield from _yield_array(array, axis)
+    elif isinstance(dataset, collections.abc.Sized):
```

Comment:
this one does not support "enumerate"?... there is some code redundancy here...

Reply:
You are rigth. `__len__` is enough to support enumerate